src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -100,8 +115,78 @@ class AmendmentTableImpl : public AmendmentTable
 void
 AmendmentTableImpl::addInitial ()
 {
-    // For each amendment this version supports, construct the AmendmentState object by calling
-    // addKnown. Set any vetoes or defaults. A pointer to the AmendmentState can be stashed
+    std::set<std::pair<uint256, std::string>> toAdd;
+
+    for (auto const& _ : detail::PreEnabledAmendments)
+    {
+        uint256 hash;
+        if (!hash.SetHex (_.first))
+        {
+            std::string const errorMsg =
+                (boost::format (
+                     "PreEnabledAmendments contains an invalid hash (expected "
+                     "a hex number). Value was: %1%") %
+                 _.first).str ();
+            throw std::runtime_error (errorMsg);
+        }
+
+        toAdd.emplace (hash, _.second);
+    }
+
+    {
+        // add the amendments from the config file
+        auto const& section = getConfig ().section (SECTION_AMENDMENTS);
+        int const numExpectedToks = 2;
+        for (auto const& _ : section.lines ())
+        {
+            boost::tokenizer<> tokenizer (_);
+            int numToks = 0;
+            uint256 id;
+            std::string friendlyName;
+            for (auto const& curTok : tokenizer)
+            {
+                ++numToks;
+                if (numToks > numExpectedToks)
+                {
+                    continue;  // continue counting so can report number in the
+                               // error
+                }
+
+                if (numToks == 1)
+                {
+                    if (!id.SetHex (curTok))
+                    {
+                        std::string const errorMsg =
+                            (boost::format (
```

Comment:
Ah, now I see why you're using `boost::format`!  Forget I made the previous comment about that.  :-)